6.0.1 - meta padding fixes

barncastle opened the pull request:
Tested against a parser, everything works now except CriteriaTree, my version has a WDBC signature but a WDB2 header - help!

Also ItemModifiedAppearance.dbd had an additional column compared to my version - can someone confirm the structure.

bloerwald commented on definitions/GroupFinderCategory.dbd:
```diff
@@ -1,15 +1,12 @@
 uint orderIndex
 
 BUILD 6.0.1.18179
-field_3184919908<32>
```
These defs come from DBMeta. Are you really sure they had the wrong column count?! Are you sure you have checked with 18179?

barncastle replied:
Yeah this is wrong. When I originally pulled the DBs from Martin's site any that failed I got the previous build.

Basically everything I've done is crap and needs redoing.

bloerwald commented on definitions/GarrFamilyName.dbd:
```diff
@@ -1,7 +1,7 @@
 COLUMNS
-uint field_2840433098?
 string field_803567487?
+int ID
```
To be technically correct, this is a guessed name ;)

bloerwald commented on definitions/GarrPlotUICategory.dbd:
```diff
@@ -1,12 +1,11 @@
 
 BUILD 6.0.1.18179
-field_329935761<32>
+$noninline,id$ID<32>
```
This is the only noninline in this diff and version. Are you sure about that?

barncastle replied:
For the record, this isn't noninline, previous build issue again.

Marlamin commented:
Dangit. I'll have to see if my extract is correct.

On Mon, 19 Mar 2018, 13:26 barncastle, <notifications@github.com> wrote:

> *@barncastle* commented on this pull request.
> ------------------------------
>
> In definitions/GarrPlotUICategory.dbd
> <https://github.com/wowdev/WoWDBDefs/pull/16#discussion_r175416769>:
>
> >  locstring categoryName_lang
>  uint field_2890979807?
>  int ID
>  uint plotType
>
>  BUILD 6.0.1.18179
> -field_329935761<32>
> +$noninline,id$ID<32>
>
> For the record, this isn't noninline, previous build issue again.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wowdev/WoWDBDefs/pull/16#discussion_r175416769>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AADEJIKAqDtGQkF1GhXBLsjQmwFeePW1ks5tf6QHgaJpZM4Svacf>
> .
>

Marlamin commented:
Looks like (in theory) my extract might be correct. 12GB zip of DBCs going from 6.0.1.18125 to 7.3.x [here](https://marlamin.com/bigdl/DBCs6xto7x.zip). Please double check if so, though. 8.0 isn't included as this is a pre-8.0 extract.

barncastle commented:
@Marlamin I think this is my issue opposed to yours. Thanks for the dump though - I'll rerun the MD5 and row compare over the next couple of days with these.

Might be worth ditching this PR and starting afresh.

bloerwald commented on definitions/GarrFollItemSet.dbd:
```diff
@@ -1,5 +1,5 @@
 COLUMNS
-uint field_458934399?
+int ID
```
Nitpicky unknown here as well

bloerwald commented on definitions/GarrGivenName.dbd:
```diff
@@ -1,9 +1,9 @@
 
 BUILD 6.0.1.18179
-field_2710791686<32>
+$id$ID<32>
```
And here. I’d either have them all as guesses or none. Your and martin’s vote which direction.

bloerwald commented on definitions/ItemModifiedAppearance.dbd:
```diff
@@ -1,10 +1,9 @@
 uint field_2799832687?
 uint field_516438342?
-uint field_3478340450?
+uint field_3478340450
```
Privably unintended.

bloerwald commented:
On “unverified ID columns”: either all `?` or none. I don’t care which way around.

barncastle commented:
- Added in Ids where there was `field_*<32>` or `padding_*<8>[4]` as the first column and where Ids were applicable i.e. used as a key in a hashmap in the client, values were unique and looked (yeah scientific I know) like the correct range and/or matched other build's known Id ranges
- Switched all Ids to unverified as per @bloerwald's suggestion (might as well start off with good intentions!) with the exception of other builds already having a known Id name or if they were gametables
- Fixed the above noted problems
- Merged matching entries together
